From fd025eeccfc7cdd687ad72ec1eaaa8f0b71117b2 Mon Sep 17 00:00:00 2001
From: Javen Ni <javen.ni@borqs.com>
Date: Fri, 26 Apr 2013 16:57:34 +0300
Subject: [PORT FROM MAIN] Telephony: Fix issues with cursor closing

BZ: 16858

There are unclosed cursor in telephony framework, which
creats issue with MMS downloading, in MTBF tests.

The fix was done by borqs in GB and later ported to ICS
and then to JB.

Change-Id: I1643b38917ae06b7d4a826e9f2a432added8e476
Orig-Change-Id: Iefe7deec1d6f07d7a461fc00f2cb7edd37865dcf
Signed-off-by: Javen Ni <javen.ni@borqs.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index ecf690a..2af362f 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -951,6 +951,10 @@ public final class DcTracker extends DcTrackerBase {
                 }
             } while (cursor.moveToNext());
         }
+        if (cursor != null) {
+            cursor.close();
+        }
+
         if (DBG) log("createApnList: X result=" + result);
         return result;
     }
-- 
1.7.4.1

